src/frontend/src/expr/table_function.rs

```diff
@@ -291,18 +294,94 @@ impl TableFunction {
         })
     }
 
-    pub fn new_postgres_query(args: Vec<ExprImpl>) -> RwResult<Self> {
-        let args = {
-            if args.len() != 6 {
-                return Err(BindError("postgres_query function only accepts 6 arguments: postgres_query(hostname varchar, port varchar, username varchar, password varchar, database_name varchar, postgres_query varchar)".to_owned()).into());
+    fn handle_postgres_or_mysql_query_args(
+        catalog_reader: &CatalogReadGuard,
+        db_name: &str,
+        schema_path: SchemaPath<'_>,
+        args: Vec<ExprImpl>,
+        expect_connector_name: &str,
+    ) -> RwResult<Vec<ExprImpl>> {
+        let cast_args = match args.len() {
+            6 => {
```

Comment:
Can we use a constant var for this? Like `INLINE_ARG_LEN` and `CONNECTION_ARG_LEN` for 6, 2 respectively